### PR TITLE
ci: Updating the publish to Codacy step to be skipped on forked PRs.

### LINKED ISSUE #14293: Skip Codacy Checks on Forked PRs
## Background

On forked PRs codacy fails

## Request(s) and Intent

Some of the checks like the Codacy reporter need an additional if condition added in order to be skipped on forked PRs (this will need to be done for any step which requires access to secrets)

## Acceptance Criteria

- [ ] Codacy is skipped on forked PRs

### PR DESCRIPTION
**Pull request by san-est**
**Description**:
Codecy reporter needs additional parameters to be skipped on forked PRs so that steps/actions which require the use of secrets are only ran on PRs from the original repo.

**Related issue(s)**:

Fixes #14293 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/node-zxc-compile-application-code.yaml [hunk 1]
```diff
@@ -494,7 +494,7 @@ jobs:
       - name: Publish to Codacy
         env:
           CODACY_PROJECT_TOKEN: ${{ secrets.codacy-project-token }}
-        if: ${{ inputs.enable-unit-tests && !cancelled() }}
+        if: ${{ inputs.enable-unit-tests && !cancelled() && github.event.pull_request.head.repo.full_name == github.repository }}
```
Comment (nathanklick) [inline .github/workflows/node-zxc-compile-application-code.yaml:497]:
```suggestion
        if: ${{ inputs.enable-unit-tests && !cancelled() &&  !github.event.pull_request.head.repo.fork }}
```

**Review by nathanklick — CHANGES_REQUESTED** [review]

File: .github/workflows/node-zxc-compile-application-code.yaml (hunk 1 above)
Comment (san-est) [reply .github/workflows/node-zxc-compile-application-code.yaml]: I've updated to the suggested change.

**Review by rbarkerSL — APPROVED** [review]
LGTM

**Review by nathanklick — DISMISSED** [review]

**Comment by san-est** [comment]
Closing pull request due to old conflicts which caused issues with the forked branch. Will open a new PR for the same.

**Comment by san-est** [comment]
New PR: #16078

**Pull request closed without merging**